packages/svelte/src/internal/client/render.js

```diff
@@ -1010,6 +1010,26 @@ export function selected(dom) {
 	});
 }
 
+/**
+ *
+ * @param {HTMLInputElement | HTMLSelectElement} element
+ * @param {(value: unknown) => void} update
+ */
+function listen_to_form_reset(element, update) {
+	// Inside effect to ensure the element is connected to the DOM
+	effect(() => {
+		const form = element.form;
```

Comment:
Forgot to mention: This is probably not fully reliable.

The associated form can dynamically be changed using the `form` attribute ([Example](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAACqWSTWrDMBCFryKGgh0wzt61AymhF-gyykK2x6lAHhlplLYY373IP2m66aZLvXnz6WlGI3TaoIfiPAKpHqGA4zBABvw1xIO_oWGEDLwNrolK6RunBz5IkmyQRWddLyrx5FkxpolKds9bifGTH0rvaIxdyiS5C9SwtiQsvYW615ziToyxIhnzweENiU_YqWA4XZiSG0veGsyNvaaLIjnxodfM2M5RkmzTMWflrsi5bu_aOc9zwg_xal1_UqzSzbS7rJ7tpnu3Q49bgElSuf95P5VG1WjmUZSahsAijq2S4FSrrQRRa2qLq7NhqMaYbhI3ZUJ0KAliP3fGLOIYyRvtH9j6N_blESupnJel2_V-S36efDVuO5jm5gdb_YdN0poufpxKQlz3Fm7JM0ZpEmvGGV0HZkurdFhw5X5RD5Igg962utPYQsEu4HSZvgGYc31spAIAAA==)).

Thought that maybe `svelte:element` shenanigans could also mess with this, but that appears to trigger effects to run again ([Example](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAACo1STWvcMBD9K4MI2AtGe3ftDYWlkHOPUQ7-GCWi8shI402C8X8vkuxtQy892TOajzfvvVVoYzGI-nkV1E0oavF9nkUl-HOOQbihZRSVCG7xQ8w0YfBm5osixWaanWd4onlh0N5NUMhzimRuLL4pioUWGbTzE7TwELhjLHVnA572Z73QwMYROPq59JPhEk-wxhfFKGePNyS-ou4Wy2VqYsWDo-AsSutey5xRXIRlMsw4pm1FdeRRcudfkaUZ77lnKSXhO_xwfrp23JVH0ellrzk23bs9BjwAbIqa8x8uqLFdjzbR0phESKSwVWJ4w-FX7z6UgN7QWKcYx3aNEDc4p5YIAtDihMRx8H0YNZnJen8EM7ZKxFYlgN9MyHPgEYp0MtRQjOZWbOAoJC7b9WB1y-iyXFHtVgnGD1ZiR9H0C7OjSy5vznuYLv2C4qJIVGJyo9EGR1GzX3Cr7hZKG_7XRFFHhhWklLN3c4AtuiT97lRH93jUWQ1S_IBa48BleYL2cvjkbzsUic6na1HFPhmJeZRm3IWLn6_SZb2SOplSj3qD9UCURfr34JftNyS28oc_AwAA)).